nifi-nar-bundles/nifi-social-media-bundle/nifi-twitter-processors/src/main/java/org/apache/nifi/processors/twitter/TweetStreamService.java

```diff
@@ -0,0 +1,222 @@
+
+    private final ScheduledExecutorService executorService;
+
+    private final Set<String> tweetFields;
+    private final Set<String> userFields;
+    private final Set<String> mediaFields;
+    private final Set<String> pollFields;
+    private final Set<String> placeFields;
+    private final Set<String> expansions;
+    private final int backfillMinutes;
+    private final TwitterApi api;
+    private InputStream stream;
+
+    private Long backoffMultiplier;
+    private int backoffAttempts;
+    private int attemptCounter;
+    private Long backoffTime;
+    private Long maximumBackoff;
+
+    private StreamEndpoint endpoint;
+
+    public TweetStreamService(final ProcessContext context, final BlockingQueue<String> queue, final ComponentLog logger) {
+        Objects.requireNonNull(context);
+        Objects.requireNonNull(queue);
+        Objects.requireNonNull(logger);
+
+        this.queue = queue;
+        this.logger = logger;
+
+        this.tweetFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.TWEET_FIELDS);
+        this.userFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.USER_FIELDS);
+        this.mediaFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.MEDIA_FIELDS);
+        this.pollFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.POLL_FIELDS);
+        this.placeFields = parseCommaSeparatedPropreties(context, ConsumeTwitter.PLACE_FIELDS);
+        this.expansions = parseCommaSeparatedPropreties(context, ConsumeTwitter.EXPANSIONS);
+        this.backfillMinutes = context.getProperty(ConsumeTwitter.BACKFILL_MINUTES).asInteger();
+
+        this.backoffMultiplier = 1L;
+        this.backoffAttempts = context.getProperty(ConsumeTwitter.BACKOFF_ATTEMPTS).asInteger();
+        this.attemptCounter = 0;
+        this.backoffTime = context.getProperty(ConsumeTwitter.BACKOFF_TIME).asTimePeriod(TimeUnit.SECONDS);
+        this.maximumBackoff = context.getProperty(ConsumeTwitter.MAXIMUM_BACKOFF_TIME).asTimePeriod(TimeUnit.SECONDS);
+
+        ApiClient client = new ApiClient();
+        final int connectTimeout = context.getProperty(ConsumeTwitter.CONNECT_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        final int readTimeout = context.getProperty(ConsumeTwitter.READ_TIMEOUT).asTimePeriod(TimeUnit.MILLISECONDS).intValue();
+        client = client.setConnectTimeout(connectTimeout);
+        client = client.setReadTimeout(readTimeout);
+        api = new TwitterApi(client);
+
+        TwitterCredentialsBearer creds = new TwitterCredentialsBearer(context.getProperty(ConsumeTwitter.BEARER_TOKEN).getValue());
+        api.setTwitterCredentials(creds);
+
+        this.executorService = Executors.newSingleThreadScheduledExecutor();
+    }
+
+    public String getBasePath() {
+        return api.getApiClient().getBasePath();
+    }
+
+    public void setBasePath(final String path) {
+        api.getApiClient().setBasePath(path);
+    }
```

Comment:
By refactoring resolution of the base path property to the constructor, both of these methods can be removed.

Reply:
Making the changes